### DISCUSSION TITLE
Cellcharter and its uses for the non-initiated for the models

### DISCUSSION THREAD

**Discussion by josenimo** [CSOgroup/cellcharter#62] [General]
Hey @marcovarrone,

I am just getting started with Cellcharter, I am really liking the approach and the idea behind it.

A simple question:
If I already cleaned my imaging data (single-cell x feature matrix), and I want to perform further spatial analyses. Do I have to perform the dimensionality reduction with the models or the raw cell x mean marker intensity table is good enough?

In your opinion what would be the use case of training my own model? For a new experiment with a decent number of cells? I am kinda lost without defaults, and would appreciate some sanity checks :)

**Comment by marcovarrone** [comment]
Hi @josenimo, thank you very much!

I think you can get a lot of information from this issue #54.

There are two reasons for using trVAE (for spatial proteomics) or scVI (for spatial transcriptomics):
- **Batch effect removal**: If you see samples that should have some similarity having completely disjoint niches, maybe there are some batch effects. But if you don't see batch effects, you don't need to do dimensionality reduction.
- **Computational efficiency**: especially from spatial transcriptomics for which you have thousands of features, clustering them will take a long time and not be very accurate. This is especially important if you are running `tl.ClusterAutoK`, which repeats the clustering many times. In this case, the time saved in all clustering runs may be bigger than the time spent training the dimensionality reduction. However, if you have 10s or around 100 features (as you often have for spatial proteomics), then also in this case the dimensionality reduction may not be necessary.

Of course, never use the trained models that I use in the tutorials for your own data as they are not meant to be generalizable :)

**Comment by LukasHats** [comment]
Just for the sake of completeness, I am posting my training script, because that was also something @josenimo was interested in, so I thought I'd make it public:

```
adata = ad.read_h5ad("path/to/anndata")

# Saling performed on the complete dataset, NOT per image anymore
sc.pp.scale(adata)

adata.X = adata.X.astype(np.float32).copy()
condition_key = 'patient_ID'
cell_type_key = 'Phenotype3'
conditions = adata.obs[condition_key].unique().tolist()


trvae_epochs = 500
surgery_epochs = 500

early_stopping_kwargs = {
    "early_stopping_metric": "val_unweighted_loss",
    "threshold": 0,
    "patience": 20,
    "reduce_lr": True,
    "lr_patience": 13,
    "lr_factor": 0.1,
}
trvae = cc.tl.TRVAE(
    adata=adata,
    condition_key=condition_key,
    conditions=conditions,
    recon_loss='mse',
    use_mmd=False,
)
trvae.train(
    n_epochs=trvae_epochs,
    alpha_epoch_anneal=200,
    early_stopping_kwargs=early_stopping_kwargs,
    enable_progress_bar=True,
    
)
trvae.save('trvae', overwrite=True)
```

BE aware that because of the size of my dataset, I could not use the pytorch trainer with `mps` but had to train using a CPU. If you want to train with a CPU, you will most likely have to set `use_mmd=False` because otherwise it will take ages (see also [here](https://github.com/theislab/scarches/issues/101))

Happy training!
